python-package/lightgbm/basic.py

```diff
@@ -1245,12 +1245,12 @@ def __getstate__(self):
         return this
 
     def __setstate__(self, state):
-        model_str = state.get('handle', None)
+        model_str = state.get('handle', None)  # list of str
```

Comment:
__setstate__ is the opposite method of __getstate__, which has been changed too. ''.join(model_str) equals to model_str, so it actually does nothing. You can close this pr.